src/PresentationalComponents/RulesTable/RulesTable.js

```diff
@@ -126,14 +128,37 @@ const RulesTable = (props) => {
     }, []);
 
     useEffect(() => {
-        fetchRules({
-            ...filters,
-            offset,
-            limit,
-            impacting,
-            sort
+        if (history.location.search) {
+            const searchParams = new URLSearchParams(history.location.search);
```

Comment:
I think that this could be simplified with reduce function right? Also it might be better to use `JSON.parse` to get real value from params.
```JS
const searchParams = new URLSearchParams(history.location.search);
const paramsObject = Array.from(searchParams).reduce((acc, [key, value]) => ({
  ...acc,
  [key]: JSON.parse(value)
}), {});
setImpacting(paramsObject.impacting);
setFilters({ ...paramsObject });
```

Reply:
this is awesome! ahhhhh had a feeling reduce coulda been used but couldn't quite get there in the heat of the moment, Json.parse won't quite work for the number strings (which gotta stay strings) but thats a great solution for pulling out the booleans!